### PR TITLE
Add ability to remove created files and dirs on exit

### PR DESCRIPTION
**Pull request by valfirst**
We use `ci-sauce` as dependency in our framework, having it we enable easy management of SC binaries for end users. The library doesn't allow to clean up created file resources and users asks/complains about them (especially when multiple SC versions are used). This PR aims to introduce missing clean up functionality.

### CODE REVIEW

File: src/test/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManagerTest.java [hunk 1]
```diff
@@ -119,35 +133,39 @@ public void openConnectionTest_existing_tunnel() throws Exception {
...
 
     @Test
-    public void testExtractZipFile() throws Exception {
+    public void testExtractZipFileWithoutCleanUpOnExit() throws Exception {
+        testExtractZipFile(false);
+    }
+
+    @Test
+    public void testExtractZipFileWithCleanUpOnExit() throws Exception {
+        testExtractZipFile(false);
```
Comment (adambabik) [inline src/test/java/com/saucelabs/ci/sauceconnect/SauceConnectFourManagerTest.java:142]: Shouldn't it be 
```suggestion
        testExtractZipFile(true);
```

The fact of removing the particular files should be tested as well.

**Comment by adambabik** [comment]
@valfirst Thank you for sending this PR. 

Can you update the description with a problem you're trying to solve, please?

**Comment by valfirst** [comment]
@adambabik could you please review the changes?

**Comment by adambabik** [comment]
@valfirst thank you, LGTM!

**Pull request merged**